delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/plugin/Plugin.scala

```diff
@@ -0,0 +1,31 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.plugin
+
+import akka.http.scaladsl.server.Route
+import monix.bio.Task
+
+/**
+  * Plugin API.
+  */
+trait Plugin {
+
+  /**
+    * Start the plugin.
+    *
+    * @param deps  plugin dependencies
+    * @return optional Akka Http [[Route]], if the plugin defines one.
+    */
+  def start(deps: PluginDependencies): Task[Option[Route]]
```

Comment:
I don't know which library is best suited to inject dependencies to the plugin but I think we will have to use one.
For the return type, besides the route, we could return the plugin id, the plugin version, a healthcheck function, maybe a retry strategy and more